docs/manual.rst

```diff
@@ -2373,13 +2374,35 @@ add or remove geometries.
   >>> from shapely.strtree import STRtree
   >>> points = [Point(i, i) for i in range(10)]
   >>> tree = STRtree(points)
-  >>> tree.query(Point(2,2).buffer(0.99))
-  >>> [o.wkt for o in tree.query(Point(2,2).buffer(0.99))]
+  >>> query_geom = Point(2,2).buffer(0.99)
+  >>> tree.query(query_geom)
+  >>> [o.wkt for o in tree.query(query_geom)]
   ['POINT (2 2)']
-  >>> [o.wkt for o in tree.query(Point(2,2).buffer(1.0))]
+  >>> query_geom = Point(2,2).buffer(1.0)
+  >>> [o.wkt for o in tree.query(query_geom)]
   ['POINT (1 1)', 'POINT (2 2)', 'POINT (3 3)']
+  >>> [o.wkt for o in tree.query(query_geom) if o.intersects(query_geom)]
+  ['POINT (2 2)']
+  
+.. note::
 
-
+   STRtree does not allow the storage of e.g. a list index number referencing the
+   position of the geometry-to-be-added in a list, along-side the geometry objects
+   like the `rtree`_ module would do. You can however, as a workaround, use
+   `monkey patching`_ to add attributes to the geometry objects themselves before
+   adding them to the STRtree.
+   
+      >>> from copy import deepcopy
+      >>> points_numbered = []
+      >>> for i, point in enumerate(points):
+      >>>     point = deepcopy(point)  # let's not mutate the originals
+      >>>     point.i = i
+      >>>     points_numbered.append(point)
+      >>> tree = STRtree(points_numbered)
+      >>> result = tree.query(query_geometry)
+      >>> [(o.i, o.wkt) for o in result]
+      [(1, 'POINT (1 1)'), (2, 'POINT (2 2)'), (3, 'POINT (3 3)')]
+      
```

Comment:
@kannes I think this is a fine example and a necessary one because shapely objects cannot currently be hashed and so can't serve as keys for an auxiliary dict. This is not quite the same thing as "monkey patching", so let's just say "attributes can be added to shapely objects and they will be saved in the STRtree.

Reply:
I think I got that term from you a long time ago, I have no strong feelings about it (though monkeys in trees are a nice sight). It feels a bit ugly to do things like this, so having a "ugly" name for the process seemed nice to me.